sdk/template/template/test/internal/beforeAfter.spec.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { InternalClass } from "../../src/internalClass";
+import { assert } from "chai";
+import { isNode } from "@azure/core-http";
+import * as fs from "fs";
+
+describe("before, beforeEach, after and afterEach examples", function() {
+  // We recommend using `beforeEach` rather than `before`,
+  // just as much as we recommend using `afterEach` rather than `after`.
+  // The idea is that each test case should not depend on a state that is shared with other tests.
+
+  // Use `beforeEach` to execute tasks that will prepare the resources needed
+  // for each test to run cleanly, and `afterEach` to tear down or clean
+  // those settings before the next test runs.
+  describe("Encouraged example of `beforeEach` and `afterEach`", function() {
+    let client: InternalClass;
+    let state: {
+      fruits?: string[];
+    } = {};
+
+    beforeEach(function() {
+      client = new InternalClass();
+      state = {
+        fruits: []
+      };
+
+      // And other per-test setups...
+    });
+
+    afterEach(function() {
```

Comment:
Maybe to illustrate how to use beforeEach and afterEach, you can open a connection in beforeEach and close it in afterEach?

Reply:
Can I use mockFs instead? Since I'm already using it in this test.

Reply:
Yes that would be good 😄

Reply:
Done!